### PR TITLE
Add the Relationship Extraction service.

### PR DESCRIPTION
**Pull request by sccheng**
### Summary

This pull request adds the Relationship Extraction service to the iOS SDK. This addresses issue #314.

### CODE REVIEW

File: Source/RelationshipExtractionV1/Models/Doc.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public let relations: Relations
+    
+    /// Used internally to initialize a `Document` model from JSON.
+    public init(json: JSON) throws {
+        id = try json.string("id")
+        text = try json.string("text")
+        sentences = try json.arrayOf("sents", "sent", type: Sentence.self)
+        mentions = try json.arrayOf("mentions", "mention", type: Mention.self)
+        entities = try json.arrayOf("entities", "entity", type: Entity.self)
+        relations = try json.decode("relations")
```
Comment (vherrin) [inline Source/RelationshipExtractionV1/Models/Doc.swift]: update .decode then the path type

File: Source/RelationshipExtractionV1/Models/Entity.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    public let entityClass: EntityClass
+    
+    /// The level of the entity. Should be the same as the `mentionType` of its primary mention.
+    public let level: EntityLevel
+    
+    /// Context-free subtype of the entity.
+    public let subtype: String
+    
+    /// An internal confidence level for the accuracy of the entity annotation, ranging from 0 to 1.
+    public let score: Double
```
Comment (vherrin) [inline Source/RelationshipExtractionV1/Models/Entity.swift]: remove

File: Source/RelationshipExtractionV1/Tests/RelationshipExtractionTests.swift [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    
+    // MARK: - Test Configuration
+    
+    override func setUp() {
+        super.setUp()
+        continueAfterFailure = false
+        instantiateRelationshipExtraction()
+    }
+    
+    /** Instantiate Text to Speech instance. */
```
Comment (glennrfisher) [inline Source/RelationshipExtractionV1/Tests/RelationshipExtractionTests.swift]: Can change `Text to Speech` to `Relationship Extraction` ;)

File: Source/RelationshipExtractionV1/Models/Mention.swift [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    /// The class of this mention.
+    public let mentionClass: MentionClass
+    
+    /// The confidence level for the accuracy of the mention annotation, on a range from 0 to 1.
+    public let score: Double
+    
+    /// The confidence level for the accuracy of a mention that coreferences, or refers to, another
+    /// mention, on a range from 0 to 1. Mentions that don't refer to other mentions will always
+    /// have a corefScore of 1.
+    public let corefScore: Double
```
Comment (vherrin) [inline Source/RelationshipExtractionV1/Models/Mention.swift]: look at java sdk to see if the internals are there

File: Source/RelationshipExtractionV1/Tests/RelationshipExtractionTests.swift [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            "ie-en-news",
+            text: "The president’s trip was designed to reward Milwaukee for its success in signing " +
+                "up people for coverage. It won a competition called the Healthy Communities " +
+                "Challenge that involved 20 cities.",
+            failure: failWithError) { document in
+            
+            XCTAssertEqual(document.entities.count, 7)
+            XCTAssertEqual(document.mentions.count, 7)
+            XCTAssertEqual(document.relations.relations.count, 1)
+            XCTAssertEqual(document.sentences.count, 2)
```
Comment (glennrfisher) [inline Source/RelationshipExtractionV1/Tests/RelationshipExtractionTests.swift]: Might be helpful to use `.responseString { response in print(response) }` to print the JSON response and then use `XCTAssert` statements to make sure that everything in the JSON is populated in the models.

File: Source/RelationshipExtractionV1/RelationshipExtraction.swift [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+     Analyzes a piece of text and extracts the different entities, along with the relationships that 
+     exist between those entities.
+     
+     - parameter language: Identifier for the language of the text. For example, "ie-en-news" for 
+            English, and "ie-es-news" for Spanish.
+     - parameter text: The text to be analyzed.
+     - parameter failure: A function executed if an error occurs.
+     - parameter success: A function executed with a Document object.
+     */
+    public func getRelationships(
```
Comment (vherrin) [inline Source/RelationshipExtractionV1/RelationshipExtraction.swift]: check java sdk for name just to see if we are close

File: Source/RelationshipExtractionV1/Tests/RelationshipExtractionTests.swift (hunk 3 above)
Comment (sccheng) [inline Source/RelationshipExtractionV1/Tests/RelationshipExtractionTests.swift]: Ohhh great catch xD

**Comment by glennrfisher** [comment]
Cool! This looks good to me. I think it's ready to be merged.

**Comment by glennrfisher** [comment]
We need to recreate the target before merging to master.

**Pull request merged**